bundles/org.openhab.binding.blink/src/main/java/org/openhab/binding/blink/internal/handler/AccountHandler.java

```diff
@@ -0,0 +1,402 @@
+            for (BlinkEvents.Media mediaEvent : events.media) {
+                if (mediaEvent.deleted) {
+                    eventStore.remove(mediaEvent.id);
+                } else {
+                    eventStore.put(mediaEvent.id, mediaEvent);
+                }
+                if (nextEventSince.isBefore(mediaEvent.updated_at)) {
+                    nextEventSince = mediaEvent.updated_at;
+                }
+                // only trigger events if not polling for the first time
+                if (!eventSince.isEqual(EPOCH_UTC)) {
+                    fireMediaEvent(mediaEvent);
+                }
+            }
+            eventSince = nextEventSince;
+        } catch (IOException e) {
+            setOffline(e);
+        }
+    }
+
+    public final @Nullable BlinkAccount getBlinkAccount() {
+        return this.blinkAccount;
+    }
+
+    public final AccountConfiguration getConfiguration() {
+        return this.config;
+    }
+
+    public final String getGeneratedClientId() {
+        return generatedClientId;
+    }
+
+    BlinkCamera getCameraState(CameraConfiguration camera, boolean refresh) throws IOException {
+        Long cameraId = camera.cameraId;
+        BlinkHomescreen devices = getDevices(refresh);
+        if (blinkAccount == null) {
+            logger.error("Blink Account not set in bridge");
+            throw new IOException("Blink Account not set in bridge");
+        }
+        if (devices == null) {
+            throw new IOException("No cameras found for account");
+        }
+        List<BlinkCamera> cameras = (camera.cameraType == null
+                || camera.cameraType == CameraConfiguration.CameraType.CAMERA) ? devices.cameras : devices.owls;
+        if (cameras == null || cameras.isEmpty()) {
+            logger.error("Unknown camera {} for account {}", cameraId, blinkAccount.account.account_id);
+            throw new IOException("No cameras found for account");
+        }
+        List<BlinkCamera> foundCameras = cameras.stream().filter(c -> Objects.equals(c.network_id, camera.networkId))
+                .filter(c -> Objects.equals(c.id, cameraId)).collect(Collectors.toUnmodifiableList());
+        if (foundCameras.size() > 1) {
+            logger.error("More than one camera {} for account {}", cameraId, blinkAccount.account.account_id);
+            throw new IOException("More than one camera found for id " + cameraId);
+        } else if (foundCameras.isEmpty()) {
+            logger.error("Unknown camera {} for account {}", cameraId, blinkAccount.account.account_id);
+            throw new IOException("Unknown camera");
+        }
+        return foundCameras.get(0);
+    }
+
+    BlinkNetwork getNetworkState(String networkId, boolean refresh) throws IOException {
+        BlinkHomescreen devices = getDevices(refresh);
+        if (blinkAccount == null) {
+            logger.error("Blink Account not set in bridge");
+            throw new IOException("Blink Account not set in bridge");
```

Comment:
You could log the exception where it is caught finally to avoid all the loggings here.